src/dtype.rs

```diff
@@ -57,21 +68,6 @@ impl PyArrayDescr {
         self.into_ptr() as _
     }
 
-    /// Returns the internal `PyType` that this `dtype` holds.
-    ///
-    /// # Example
-    /// ```
-    /// pyo3::Python::with_gil(|py| {
-    ///    let array = numpy::PyArray::from_vec(py, vec![0.0, 1.0, 2.0f64]);
-    ///    let dtype = array.dtype();
-    ///    assert_eq!(dtype.get_type().name().unwrap().to_string(), "float64");
-    /// });
-    /// ```
-    pub fn get_type(&self) -> &PyType {
```

Comment:
Should we consider marking as deprecated instead of removing immediately, to make it easier for users to migrate? It may not be immediately obvious what they should migrate to.

Reply:
That crossed my mind as well - sure, we can leave is as deprecated for a version or so.

Reply:
👍 thanks, can even suggest what to migrate to in the deprecation note!